client/src/classes/undo-redo-commands/ComplexCommand.ts

```diff
@@ -0,0 +1,36 @@
+import { UndoRedoCommand } from "@/interfaces/undo-redo-context";
+
+/**
+ * The class is used in the canvas undo/redo functionality.
+ * It is used to aggregate multiple objects of UndoRedoCommand into one such a command.
+ *
+ * @param commands an array of commands to perform. They should be in the same order as they should be performed during the **redo** method.
+ */
+export class ComplexCommand implements UndoRedoCommand {
```

Comment:
I don't think I got the naming right with 'ComplexCommand.' It's not very clear and could imply many things. Since it's primarily related to the aggregation of multiple objects, maybe 'GroupObjectCommand' would be a better fit? I'm not entirely sure... If you can't think of a better name, feel free to leave it as it is.

Reply:
What I meant to convey here by naming this class `ComplexCommand` is that we use the composite design pattern.
https://refactoring.guru/design-patterns/composite
So the sole purpose of `ComplexCommand` is to aggregate multiple commands.

GroupObjectCommand sounds way to fabrick-ish, as if it had something to do with fabric's objects or fabric's groups. Other names I can propose are `CompositeCommand`, `CompoundCommand` or `AggregateCommand`.

Reply:
👍